core/src/main/java/com/datastax/oss/driver/api/core/cql/PreparedStatement.java

```diff
@@ -58,6 +59,13 @@ public interface PreparedStatement {
   @NonNull
   ColumnDefinitions getVariableDefinitions();
 
+  /**
+   * Table name inferred from {@link PreparedStatement#getVariableDefinitions()} or {@code null} if
+   * not possible.
+   */
+  @Nullable
+  CqlIdentifier inferTable();
+
```

Comment:
Main motivation was keeping new additions as specific as possible and not extending what I don't need to. `PreparedStatement` has 1 implementation while `Request` has 20 implementations. At first I thought it will be easier to maintain this way, but I may as well provide default implementation that returns null which should not complicate writing implementations of Request.

Routing table can be introduced. I probably should add field and setters too in that case.

Reply:
Moved this method to `Request` as `getRoutingTable` and provided override for `BatchStatement` and `BoundStatement`